### PR TITLE
Allow auto-update of rules for invalid gears [FLYW-1315]

### PR DESCRIPTION
**Pull request by ehlertjd**
### Review Checklist

- Tests were added to cover all code changes
- Is there a DB upgrade or fix?
  - If so is it covered by integeration tests and run against dev?
- Would this PR benefit from a test plan? If so, is one provided?
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

**Comment by nagem** [comment]
With this solution you can't "roll back" a gear, so that is something to note. For example:
  - Gear rules are on gear v.1 with auto-update turned on
  - Gear v.2 rolls out, gear rules are updated
  - Gear v.2 is deemed problematic and disabled
  - Gear rules are manually brought back to Gear v.1
  - Gear v.3 rolls out, gear rules are not updated as the find will return Gear v.2

**Comment by nagem** [comment]
Can the logic be changed to look for gear rules with a specific gear alg name or is that not stored on the gear rule?

**Comment by ehlertjd** [comment]
@nagem I've adjusted the logic to update *any* rule on a prior gear version (with auto-update enabled), not just most-recent.

File: tests/integration_tests/python/test_rules.py [hunk 1]
```diff
@@ -859,6 +859,78 @@ def test_auto_update_rules(data_builder, api_db, as_admin):
     assert r.ok
 
 
+def test_auto_update_invalid_rule(data_builder, api_db, as_admin):
```
Comment (ryandavisflywheel) [inline tests/integration_tests/python/test_rules.py]: Unit test!  ... very nice.  This makes it easier to understand this PR's aim.

File: tests/integration_tests/python/test_rules.py [hunk 2]
```diff
@@ -859,6 +859,78 @@ def test_auto_update_rules(data_builder, api_db, as_admin):
...
+        'any': [],
+        'not': [],
+        'all': [{'type': 'file.type', 'value': 'tabular data'}],
+        'disabled': False,
+        'auto_update': True
+    })
+    assert r.ok
+    rule2_id = r.json()['_id']
+
+    # Should auto-update all rules to gearv2
```
Comment (ryandavisflywheel) [inline tests/integration_tests/python/test_rules.py]: This comment's assertion doesn't actually gets asserted... maybe a GET rules and assert gearv2?

File: api/jobs/gears.py [hunk 3]
```diff
@@ -222,8 +223,8 @@ def check_for_gear_insertion(doc):
     if conflict is not None:
         raise Exception('Gear "' + doc['gear']['name'] + '" version "' + doc['gear']['version'] + '" already exists, consider changing the version string.')
 
-def auto_update_rules(gear_id, last_gear_id):
-    config.db.project_rules.update_many({'gear_id': str(last_gear_id), 'auto_update': True}, {"$set": {'gear_id': str(gear_id)}})
+def auto_update_rules(gear_id, installed_gear_ids):
+    config.db.project_rules.update_many({'gear_id': {'$in': installed_gear_ids}, 'auto_update': True}, {"$set": {'gear_id': str(gear_id)}})
```
Comment (ryandavisflywheel) [inline api/jobs/gears.py]: ... some new MongoDB syntax to me here... cool :)

**Pull request merged**